### PR TITLE
RDK-53549: Blocklist Flag Set and Get in systemservice

### PR DESCRIPTION
**Pull request by santoshcomcast**
Blocklist Flag Set and Get in systemservice.

### CODE REVIEW

File: SystemServices/System.json [hunk 1]
```diff
@@ -1973,6 +1988,44 @@
                 "type": "string",
                 "example": "DO_NOT_SHARE"
             }
+        },
+        "getBlocklistFlag":{
+            "summary": "get the blocklist falg on device for device state. Valid flags are:  \n* `true` - The device is in blocked state.  \n* `false` - The device is in un-blocked state.",
```
[the remark this answers is not in the document]
Comment (santoshcomcast) [reply SystemServices/System.json]: Corrected.

File: SystemServices/System.json [hunk 2]
```diff
@@ -1973,6 +1988,44 @@
...
+            "result": {
+                "type": "object",
+                "properties": {
+                    "blocklist": {
+                        "summary": "blocklist flag",
+                        "type": "boolean",
+                        "example": true
+                        },
+                    "success": {
```
[the remark this answers is not in the document]
Comment (santoshcomcast) [reply SystemServices/System.json]: Corrected "success" details.

File: SystemServices/System.json [hunk 3]
```diff
@@ -1973,6 +1988,44 @@
...
+                            "summary": "returns true or false to indicate if the operation was successful or not",
+                            "type": "boolean",
+                            "example": true
+                        }
+                    }
+                }
+        },
+        "setBlocklistFlag":{
+            "summary": "Sets the blocklist falg on device for device state. Valid flags are:  \n* `true` - The device is in blocked state.  \n* `false` - The device is in un-blocked state.",
```
[the remark this answers is not in the document]
Comment (santoshcomcast) [reply SystemServices/System.json]: Corrected.

File: SystemServices/SystemServices.cpp [hunk 4]
```diff
@@ -100,6 +100,8 @@ using namespace std;
 
 #define PRIVACY_MODE_FILE "/opt/secure/persistent/System/privacymode.txt"
 
+#define DEVICESTATE_FILE "/opt/secure/persistent/opflashstore/devicestate.txt"
```
[the remark this answers is not in the document]
Comment (santoshcomcast) [reply SystemServices/SystemServices.cpp]: Created the separate definitions for directory, file, and "blocklist" parameter strings.

File: SystemServices/SystemServices.cpp [hunk 5]
```diff
@@ -1374,6 +1379,274 @@ namespace WPEFramework {
 		}
                 returnResponse(status);
         }
+        
+        bool checkOpFlashStoreDir()
+        {
+            int ret = 0;
+            if (access("/opt/secure/persistent/opflashstore", F_OK) == -1) {
+                ret = mkdir("/opt/secure/persistent/opflashstore", 0774);
```
[the remark this answers is not in the document]
Comment (santoshcomcast) [reply SystemServices/SystemServices.cpp]: Updated with new definition name.

File: SystemServices/SystemServices.cpp [hunk 6]
```diff
@@ -1374,6 +1379,274 @@ namespace WPEFramework {
...
+                LOGWARN(" --- SubDirectories created from mkdir %d ", ret);
```
[the remark this answers is not in the document]
Comment (santoshcomcast) [reply SystemServices/SystemServices.cpp]: Corrected.

**Review by ramasat — APPROVED** [review]

**Review by ramasat — APPROVED** [review]

**Pull request merged**